### PR TITLE
bash-completion: fix for sed command

### PR DESCRIPTION
**Pull request by peshay**
Prevent MacOS X BSD sed command to remove leading "t" from .ssh/config Host entries.
For example currently this host entry
`Host tas-first-letter`
will complete to 
`ssh as-first-letter`
My change will prevent this.

### CODE REVIEW

File: Library/Formula/bash-completion.rb [hunk 1]
```diff
@@ -69,7 +69,7 @@ def caveats; <<-EOS.undent
      # append any available aliases from config files
      if [[ ${#config[@]} -gt 0 && -n "$aliases" ]]; then
 -        local hosts=$( sed -ne 's/^[ \t]*[Hh][Oo][Ss][Tt]\([Nn][Aa][Mm][Ee]\)\{0,1\}['"$'\t '"']\{1,\}\([^#*?]*\)\(#.*\)\{0,1\}$/\2/p' "${config[@]}" )
-+        local hosts=$( sed -ne 's/^['"$'\t '"']*[Hh][Oo][Ss][Tt]\([Nn][Aa][Mm][Ee]\)\{0,1\}['"$'\t '"']\{1,\}\([^#*?]*\)\(#.*\)\{0,1\}$/\2/p' "${config[@]}" )
++        local hosts=$( sed -ne 's/^['$'\t '']*[Hh][Oo][Ss][Tt]\([Nn][Aa][Mm][Ee]\)\{0,1\}['$'\t '']\{1,\}\([^#*?]*\)\(#.*\)\{0,1\}$/\2/p' "${config[@]}" )
```
Comment (mikemcquaid) [inline Library/Formula/bash-completion.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (peshay) [inline Library/Formula/bash-completion.rb]: Sorry I'm  a bit new to all this. With upstream developers, do you mean scop/bash-completion ?
They are already on versio 2.2 while this code here seems to be based on 1.3
This line looks at their project different anyway:         
`local hosts=$( sed -ne 's/^[ \t]*[Hh][Oo][Ss][Tt]\([Nn][Aa][Mm][Ee]\)\{0,1\}['"$'\t '"']\{1,\}\([^#*?]*\)\(#.*\)\{0,1\}$/\2/p' "${config[@]}" )`
I already tried to make a pull request on the 2.2 version: https://github.com/scop/bash-completion/pull/17
But even if they would change the 1.3 and make a new 1.4, wouldn't this patch line here stop to work?

Comment (mikemcquaid) [inline Library/Formula/bash-completion.rb]: I believe upstream is https://bash-completion.alioth.debian.org/. We just want to make sure they know this change is needed.

Comment (peshay) [inline Library/Formula/bash-completion.rb]: The upstream developers made a slightly different change:
https://github.com/scop/bash-completion/commit/657fcf62c77343ede3a21430a6c8e4faf4cd1dbe

**Comment by peshay** [comment]
duplicate from https://github.com/Homebrew/homebrew/pull/50422

**Comment by peshay** [comment]
duplicate from https://github.com/Homebrew/homebrew/pull/50422

**Pull request closed without merging**